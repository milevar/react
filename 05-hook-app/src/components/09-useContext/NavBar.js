import React from 'react';
import {Link, NavLink} from "react-router-dom";

const NavBar = () => {
    const activeClassName = 'nav-link active';
    const inactiveClassName = 'nav-link';
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">
                <Link to="/" className="navbar-brand">useContext</Link>
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <NavLink to="/"
                                 className={({isActive}) => isActive ? activeClassName : inactiveClassName}>
                            Home
                        </NavLink>
                        <NavLink to="/login"
                                 className={({isActive}) => isActive ? activeClassName : inactiveClassName}>
                            Login
                        </NavLink>
                        <NavLink to="/about"
                                 className={({isActive}) => isActive ? activeClassName : inactiveClassName}>
                            About
                        </NavLink>
                    </div>
                </div>
            </div>
        </nav>
    );
};

export default NavBar;
