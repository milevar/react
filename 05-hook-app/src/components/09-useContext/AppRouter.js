import React from 'react';
import {BrowserRouter, Route, Routes} from 'react-router-dom';
import NavBar from './NavBar';
import HomeScreen from './HomeScreen';
import AboutScreen from './AboutScreen';
import LoginScreen from './LoginScreen';

const AppRouter = () => {
    return (
        <BrowserRouter>
            <NavBar/>

            <div className="container">
                <Routes>
                    <Route path="/" element={<HomeScreen/>}/>
                    <Route path="/about" element={<AboutScreen/>}/>
                    <Route path="/login" element={<LoginScreen/>}/>
                    <Route path="*" element={<HomeScreen/>}/>
                </Routes>
            </div>
        </BrowserRouter>
    );
};

export default AppRouter;
