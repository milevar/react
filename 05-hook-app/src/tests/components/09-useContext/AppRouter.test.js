import React from "react";
import { mount } from 'enzyme';

import AppRouter from "../../../components/09-useContext/AppRouter";
import {UserContext} from "../../../components/09-useContext/UserContext";

describe('Pruebas en <AppRouter />', () => {

    const user = {
        name: 'Milena',
        email: 'milena@gmail.com'
    }
    const wrapper = mount(
        <UserContext.Provider value={{ user }}>
            <AppRouter />
        </UserContext.Provider>
    );

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

});
