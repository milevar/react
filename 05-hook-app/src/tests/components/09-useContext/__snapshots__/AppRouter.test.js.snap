// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
          <a
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "name": "Milena",
   "email": "milena@gmail.com"
}
      </pre>
    </div>
  </div>,
]
`;
